### ISSUE TITLE
E1041/E1040 false positive: Ref/GetAtt on AWS::CertificateManager::Certificate rejected for AcmCertificateArn destinations

### ISSUE THREAD

**Issue by maciejstromich** [aws-cloudformation/cfn-lint#4642]
## cfn-lint version
```
cfn-lint 1.55.1
Python 3.14.7
Linux (aarch64), via the official cfn-lint pip package
```

## Description

`E1041`/`E1040` fire as false positives for `AWS::CertificateManager::Certificate` whenever its `Ref` or `GetAtt CertificateArn` is used to populate any property whose destination format is `AWS::ACM::Certificate.Arn` (e.g. `AWS::CloudFront::Distribution`'s `ViewerCertificate.AcmCertificateArn`).

This only reproduces when cfn-lint pulls the auto-downloaded "enhanced schemas" snapshot from `aws-cloudformation/resource-provider-enhanced-schemas` (the schema referenced by `_ENHANCED_SCHEMAS_URL`/`_VERSION_URL` in `cfnlint/schema/manager.py`) instead of the schema bundled with the pip package — see root cause below.

## Steps to reproduce

Minimal template (`repro.yaml`):

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  Certificate:
    Type: AWS::CertificateManager::Certificate
    Properties:
      DomainName: example.com
      ValidationMethod: DNS
  CloudFrontDistribution:
    Type: AWS::CloudFront::Distribution
    Properties:
      DistributionConfig:
        Enabled: true
        DefaultCacheBehavior:
          TargetOriginId: origin1
          ViewerProtocolPolicy: redirect-to-https
        Origins:
          - Id: origin1
            DomainName: example.com
            CustomOriginConfig:
              OriginProtocolPolicy: https-only
        ViewerCertificate:
          AcmCertificateArn: !Ref Certificate
          SslSupportMethod: sni-only
          MinimumProtocolVersion: TLSv1.2_2021
```

```
$ cfn-lint repro.yaml
E1041 {'Ref': 'Certificate'} does not match destination format of 'AWS::ACM::Certificate.Arn'
repro.yaml:22:11
```

Swapping to `!GetAtt Certificate.CertificateArn` reproduces the sibling check instead:

```
E1040 {'Fn::GetAtt': ['Certificate', 'CertificateArn']} does not match destination format of 'AWS::ACM::Certificate.Arn'
```

Both are the AWS-documented ways to obtain the certificate ARN for this resource (`Ref` returns the ARN for `AWS::CertificateManager::Certificate` per AWS docs; `CertificateArn` is also listed in `readOnlyProperties`), so there is no way to reference the certificate here without tripping one of the two rules.

## Root cause

cfn-lint prefers whichever of the bundled vs. auto-downloaded (`~/.cache/aws/cfn-lint/schemas`) schema sets has the newer `schema_date` (`ProviderSchemaManager._resolve_schema_dirs`, `cfnlint/schema/manager.py`). In our run:

- Bundled package schema: `schema_date: 2026-08-11T22:19:36Z` — **no false positive**
- Auto-downloaded schema (from `resource-provider-enhanced-schemas` "latest" release): `schema_date: 2026-08-18T08:13:35Z` — **triggers E1041/E1040**

Diffing the `AWS::CertificateManager::Certificate` resource schema between the two sets, the *only* difference is that the `format` keyword was dropped from the `CertificateArn` property in the newer snapshot:

```
OLD properties.CertificateArn:
{"description": "...", "format": "AWS::ACM::Certificate.Arn", "maxLength": 2048, "minLength": 20, "pattern": "...", "type": "string"}

NEW properties.CertificateArn:
{"description": "...", "maxLength": 2048, "minLength": 20, "pattern": "...", "type": "string"}
```

Note that a separate, synthetic `properties.Id` entry (used by cfn-lint internally, not a real CloudFormation attribute) still carries the format unchanged in both snapshots:

```
properties.Id (unchanged in both): {"format": "AWS::ACM::Certificate.Arn"}
```

`primaryIdentifier` is `["/properties/CertificateArn"]` in both snapshots, and `RefFormat`/`GetAttFormat` (`cfnlint/rules/functions/RefFormat.py`, and the GetAtt equivalent) resolve `Ref`/`GetAtt` straight through to the `CertificateArn` property definition — they never consult the sibling `Id` property. So once `format` is missing from `CertificateArn` itself, the format-compatibility comparison against `AcmCertificateArn`'s destination format (which still declares `AWS::ACM::Certificate.Arn`) fails, even though the returned value is still the correctly-formatted ARN.

## Related precedent

This is the same failure shape as a previously-fixed issue:

- [#4468](https://github.com/aws-cloudformation/cfn-lint/issues/4468) — "E1041 false positive: Lambda Alias Ref rejected where Lambda Function ARN is expected." `Ref` on `AWS::Lambda::Alias` resolved to a property that was missing the `Lambda::Function.Arn` format annotation, so it failed the same format-compatibility comparison against a destination that declared that format.
- Fixed by [#4469](https://github.com/aws-cloudformation/cfn-lint/pull/4469) — "Fix E1041 false positive: add `Lambda::Function.Arn` format to Lambda Alias" — the fix added the missing `format` annotation back as a schema extension, rather than changing the `RefFormat` rule logic.
- Separately, [resource-provider-enhanced-schemas#14](https://github.com/aws-cloudformation/resource-provider-enhanced-schemas/pull/14) — "Exclude ELBv2 Listener resources from ACM certificate format" — shows that ACM-certificate-format annotations for this exact `AWS::ACM::Certificate.Arn` format specifically are actively maintained in the `resource-provider-enhanced-schemas` repo, which is where this fix likely belongs too (as either a restored `format` on `CertificateArn`, or a cfn-lint-side extension patch following the #4469 pattern).

I also checked [#4637](https://github.com/aws-cloudformation/cfn-lint/pull/4637) ("Update CloudFormation schemas to 2026-08-18"), since its title suggested it might be the source of or fix for this regression — it isn't either. Its diff only touches `AdditionalSpecs/Policies.json` and `extensions/*` overlays for EC2 instance types, RDS engine versions/instance classes, Redshift node types, and SageMaker instance types; nothing related to `AWS::CertificateManager::Certificate`. More fundamentally, cfn-lint's repo doesn't version the base provider/resource schemas at all (`src/cfnlint/data/schemas/` on `main` has no `providers/`/`resources/` directory) — they're pulled at install/runtime from `resource-provider-enhanced-schemas`, which is where the `CertificateArn` format annotation was actually dropped.

## Expected behavior

`!Ref Certificate` / `!GetAtt Certificate.CertificateArn` should not trigger E1041/E1040 when used for a destination declaring format `AWS::ACM::Certificate.Arn`, since this is exactly the documented, correct way to obtain the certificate's ARN.

## Suggested fix

Either (or both):
1. In `resource-provider-enhanced-schemas`, restore `"format": "AWS::ACM::Certificate.Arn"` on `properties.CertificateArn` for `AWS::CertificateManager::Certificate` (this looks like an accidental drop between the `2026-08-11` and `2026-08-18` schema releases, since it's the only diff in the resource and the sibling `Id` property still has it).
2. In cfn-lint's `Ref`/`GetAtt` format resolution, when the resolved property itself has no `format`, fall back to checking other properties/pointers in the same schema that alias the same value (e.g. the synthetic `Id` property) before concluding there's a mismatch — so a single missing annotation on one property doesn't produce a hard format-mismatch error for a resource that structurally still exposes the correct format elsewhere.

## Workaround

None found that avoids both `E1041` and `E1040` simultaneously, since both intrinsics resolve through the same now-unannotated `CertificateArn` property. Current workaround is a per-resource `Metadata.cfn-lint.config.ignore_checks` suppression, or pinning cfn-lint to skip the schema auto-update (not currently exposed as a CLI flag we could find — only `-u/--update-specs` to force it).
